File: 2015/day02/Main.java
import java.io.IOException;
import java.nio.file.Files;
import java.nio.file.Paths;
import java.util.List;

public class Main {

    public static List<String> readFile(String filename) {
        try {
            List<String> content = Files.readAllLines(Paths.get(filename));
            return content;
        } catch (IOException e) {
            e.printStackTrace();
        }
        return List.of();
    }

    public static Integer part1(List<String> input) {
        String sequence = input.get(0);
        char[] chars = sequence.toCharArray();
        Integer floor = 0;
        for (char c : chars) {
            if (c == '(') {
                floor += 1;
            }
            else {
                floor -= 1;
            }
        }
        return floor;
    }

    public static Integer part2(List<String> input) {
        String sequence = input.get(0);
        char[] chars = sequence.toCharArray();
        Integer floor, index;
        floor = index = 0;
        for (char c : chars) {
            if (c == '(') {
                floor += 1;
            }
            else {
                floor -= 1;
            }
            if (floor == -1) {
                return index;
            }
            index += 1;
        }
        return floor;
    }

    public static void main(String[] args) {
        List<String> lines = readFile("input.txt");
        System.out.println("Part 1: " + part1(lines));
        System.out.println("Part 2: " + part2(lines));
    }
}

File: 2015/day02/main.py
# Part 1 ----------------------------------------------------------------------

def part1(lines):
    total = 0
    for i in lines:
        l, w, h = map(int, i.split("x"))
        area = 2*l*w + 2*w*h + 2*h*l
        smallest = min(l*w, w*h, h*l)
        total += area + smallest
    return total

# Part 2 ----------------------------------------------------------------------

def part2(lines):
    total = 0
    for i in lines:
        l, w, h = map(int, i.split("x"))
        arr = [l, w, h]
        arr.sort()
        l, w, h = arr[0], arr[1], arr[2],
        total += l*2 + w*2 + l*w*h
    return total

def main():
    filename = 'input.txt'
    file = open(filename, 'r')
    lines = file.readlines()
    lines = [line.strip() for line in lines]

    print("Part 1:", part1(lines))
    print("Part 2:", part2(lines))


if __name__ == "__main__":
    main()